### PR TITLE
feat(app-launcher): add che and workspace service for app launcher.

### PR DESCRIPTION
**Pull request by vikram-raj**
Depends on PR - https://github.com/fabric8-launcher/ngx-launcher/pull/375.

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
@vikram-raj I have two asks

- can you check if it is still working against latest `launcher-backend` and upgraded front-end
- as it's marked as `WIP` could you provide a short check-list of what is still missing?

**Comment by vikram-raj** [comment]
@bartoszmajsak I checked with latest ngx-launcher and its working fine. From my side its done. just waiting for review.

@christianvogt Is there any work required for this PR?

**Comment by invincibleJai** [comment]
@vikram-raj have verified the changes, works fine as expected.

File: src/app/space/app-launcher/services/app-launcher-che.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      providers: [
+        { provide: AppLaunchCheService, useValue: mockAppLauncherCheService }
+      ]
+    });
+    appLaunchCheService = TestBed.get(AppLaunchCheService);
+  });
+
+  it('Get che state', async () => {
+    appLaunchCheService.getState().subscribe((val: Che) => {
+      expect(val).toEqual(mockCheState);
```
Comment (bartoszmajsak) [inline src/app/space/app-launcher/services/app-launcher-che.service.spec.ts]: Not sure what's the goal of this test. Are we testing that mock is returned? Can we think about some higher level functionality/behaviour instead of making sure we know how to pass values and mocks around? :)

File: src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      providers: [
+        {provide: AppLaunchWorkSpaceService, useValue: mockAppLaunchWorkSpaceService}
+      ]
+    });
+    appLaunchWorkSpaceService = TestBed.get(AppLaunchWorkSpaceService);
+  });
+
+  it('Create workSpace', async () => {
+    appLaunchWorkSpaceService.createWorkSpace('').subscribe((val: WorkspaceLinks) => {
+      expect(val).toEqual(mockCreateWorkSpace);
```
Comment (bartoszmajsak) [inline src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts]: What is the meaning of `''`  (empty string parameter) here? Can we structure the tests in a form of functionality (and boundary cases) which this service should expose?

**Comment by joshuawilson** [comment]
@vikram-raj are you still working on this. It looks like the depending PR is merged.

File: src/app/space/app-launcher/services/app-launcher-che.service.spec.ts (hunk 1 above)
Comment (vikram-raj) [reply src/app/space/app-launcher/services/app-launcher-che.service.spec.ts]: There are already tests for ```cheService``` in fabric8-ui. You can find it here https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/services/che.service.spec.ts

File: src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts (hunk 2 above)
Comment (vikram-raj) [reply src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts]: Here is the unit test for ```workSpaceService```https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/services/workspaces.service.spec.ts.

> Can we structure the tests in a form of functionality (and boundary cases) which this service should expose?

Yes, we can write the tests in the form of functionality. But, those tests should go in ```ngx-launcher``` because the component that uses this service is in ```ngx-launcher```.

Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts]: That's fine - can you contribute those tests there and remove this one?

**Comment by bartoszmajsak** [comment]
@vikram-raj I am approving this PR, but please add tests in `ngx-launcher` as discussed in this thread https://github.com/fabric8-ui/fabric8-ui/pull/3347#discussion_r227057167

**Comment by vikram-raj** [comment]
Thanks, @bartoszmajsak. Sure I will add tests in ```ngx-launcher```

**Comment by joshuawilson** [comment]
I've created a story for the fabric8-ui work at https://openshift.io/openshiftio/Openshift_io/plan/detail/903
Please let me know when the launcher-ngx change is done and then we can follow up.

**Comment by Preeticp** [comment]
Docs have been updated via https://github.com/fabric8io/fabric8-online-docs/pull/447
